addons/account/models/account_payment_term.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, exceptions, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons import decimal_precision as dp
+
+from dateutil.relativedelta import relativedelta
+
+
+class AccountPaymentTerm(models.Model):
+    _name = "account.payment.term"
+    _description = "Payment Terms"
+    _order = "sequence, id"
+
+    def _default_line_ids(self):
+        return [(0, 0, {'value': 'balance', 'value_amount': 0.0, 'sequence': 9, 'days': 0, 'option': 'day_after_invoice_date'})]
+
+    name = fields.Char(string='Payment Terms', translate=True, required=True)
+    active = fields.Boolean(default=True, help="If the active field is set to False, it will allow you to hide the payment terms without removing it.")
+    note = fields.Text(string='Description on the Invoice', translate=True)
+    line_ids = fields.One2many('account.payment.term.line', 'payment_id', string='Terms', copy=True, default=_default_line_ids)
+    company_id = fields.Many2one('res.company', string='Company')
+    sequence = fields.Integer(required=True, default=10)
+
+    @api.constrains('line_ids')
+    @api.one
+    def _check_lines(self):
+        payment_term_lines = self.line_ids.sorted()
+        if payment_term_lines and payment_term_lines[-1].value != 'balance':
+            raise ValidationError(_('The last line of a Payment Term should have the Balance type.'))
+        lines = self.line_ids.filtered(lambda r: r.value == 'balance')
+        if len(lines) > 1:
+            raise ValidationError(_('A Payment Term should have only one line of type Balance.'))
+
+    @api.one
```

Comment:
Where we not going to stop with api.one? ;-)

Reply:
@Yenthe666 The goal is to move stuff, not to refactor the code.

Reply:
Still a good moment to touch it in master, not? You're creating a diff already so why not do two things at once 😄 Just my five cents!

Reply:
@Yenthe666 Ok ok... you won ;)

Reply:
@smetl Awesome, I own you a beer on the next OEXP 😉